assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -898,8 +897,10 @@ private Map<String, Object> submissionToMap(Set<String> activeSubmitters, Assign
 
             if (!feedbackAttachments.isEmpty()) submission.put("feedbackAttachments", feedbackAttachments);
 
-            String grade = assignmentService.getGradeForSubmitter(as, as.getSubmitters().isEmpty() ? null : as.getSubmitters().stream().findAny().get().getSubmitter());
-            if (StringUtils.isNotBlank(grade)) submission.put("grade", grade);
+            if (mergeOverride) {
+                String grade = assignmentService.getGradeForSubmitter(as, as.getSubmitters().isEmpty() ? null : as.getSubmitters().stream().findAny().get().getSubmitter());
```

Comment:
I believe line 901 is where the student override score is harnessed to then be erroneously placed in the 'Grade' field (for the overarching score for the group). The following line seals the deal.

Are you contending that as.getSubmitters().stream().findAny().get().getSubmitter() is actually affecting the data that would be passed back via JSON? At a cursory level, the (sole?) function of that chain of calls seems to be to identify a submitter (to then get that submitter's grade, etc.)

Reply:
No, what I mean is that findAny can return an empty optional and get() will throw an exception in that case. I should have handled it when I added it.

Reply:
Assuming that the only effect (and perhaps intent) of lines 901-902 is to assert as the group's grade a student override, then it would seem that lines 900-903 can be removed along with the removal of other references to 'mergeOverride'. Do you concur? If not, can you clarify what would need to be addressed in a revised commit?